tests/system_tests_tcp_adaptor.py

```diff
@@ -528,6 +530,15 @@ def router(name, mode, connection, extra=None):
                     unseen = [srv for srv in cls.router_order if "ES_" + srv not in seen]
                     cls.logger.log("TCP_TEST Router %s sees only %d of %d addresses. Waiting for %s" %
                                    (rtr, len(server_lines), len(cls.router_order), unseen))
+                if poll_loops == 1:
+                    # last poll loop
+                    for line in lines:
+                        cls.logger.log("TCP_TEST Router %s : %s" % (rtr, line))
+            poll_loops -= 1
+            if poll_loops == 0:
+                assert False, "TCP_TEST TCP_Adaptor test setup failed. Echo tests never executed."
```

Comment:
```suggestion
                self.fail("TCP_TEST TCP_Adaptor test setup failed. Echo tests never executed.")
```
https://docs.python.org/3/library/unittest.html#unittest.TestCase.fail